heron/proto/topology.proto

```diff
@@ -69,7 +75,10 @@ message Config {
     required string key = 1;
     // Only one of the below are set
     optional string value = 2;
-    optional bytes java_serialized_value = 3;
+    //optional bytes java_serialized_value = 3;
+    optional bytes serialized_value = 3;
+
+    required ConfigValueType type = 4;
```

Comment:
If these objects are just created and passed around by a running topology, that's fine to do something non-backward compatible since they just update upon redeploy. If they get persisted while running and accessed by another service, like the heron UI, then they need to be backwards compatible so the service can handle both the old and the new during the transition. I'm not sure if that's the case or not for these objects though.